app/src/components/core/dataschema/dataschema.service.ts

```diff
@@ -127,7 +127,7 @@ module app.core.dataschema {
         removeProperty(name:string, path:string[]):boolean {
             var parent = this.getFolderAt(path);
 
-            if (parent === null || !parent.hasOwnProperty('properties') || !parent.properties.hasOwnProperty(name)) {
+            if (parent === null || !parent.hasOwnProperty('properties') || !parent.properties.hasOwnProperty(name.split('/').pop())) {
```

Comment:
I would extract the last conditional into a private predicate method. It's not obvious to me what `!parent.properties.hasOwnProperty(name.split('/').pop()))` does.